docs/design/persistent-storage.md

```diff
@@ -0,0 +1,220 @@
+				GCEPersistentDisk { modes: ReadWriteOnce | ReadOnlyMany, resources: 5gb 30iops, .48 throughput  }
+			MaxInstances  100
+			IncrementBy   10
+		Status
+			InstanceCount
+			
+
+	//
+	//  addition to VolumeSource
+	//	
+	type MountMode byte
+	
+	const (
+		ReadWriteOnce MountMode = 1 << iota
+		ReadOnlyMany
+		ReadWriteMany
+	)
+	
+	func (vs *VolumeSource) canMount(mode MountMode) bool {
+		return vs.modes & mode != 0
+	}
+	
+```
+
+> To manually make storage available, the admin would, for example, create a GCE VM, create, attach, and format 20 persistent disks.  The IDs are saved and the admin then creates 20 persistent-disk.json files and POSTs each GCE_PD to the API.  There are many ways to script adding storage to a cluster.
+
+
+
+#### Pod to specify storage requirements
+
+```
+
+	Pod.Spec.Volumes.Volume 
+			Name
+			VolumeSource -- to be bound with available storage from the pool
+			PersistenceRequirements
+				Mode				ReadWriteOnce | ReadOnlyMany
+				RetentionPolicy		DeleteOnPodExit | RetainOnPodExit
+				Resources			ResourcesList (size, iops, throughput)
+				Selector			Labels to identify the PersistentStorage backing implementation
+
+```
+
+`PersistenceRequirements` are the capabilities and performance characteristics a pod author can request and rely on for the entire lifecycle of the volume.  Requests for storage that do not match available storage will go unfulfilled.
+
+
+#### Storage discovery 
+
+An API and CLI is needed to view available and in-use storage by namespace.
+
+**Use Case**:  Susie needs to crunch some data and serve it afterwards.  A single pod to create data is sufficient, but she estimates needing 10 web front ends to serve the data afterwards.  For this use case, the data does not change once created.  Susie first looks to see what is available in the cluster.
+
+```
+$ cluster/kubectl.sh get storage
+
+NAME                LABELS              MODES               RESOURCES
+_pv_vol_01          name=my-data        RWO,ROX             size=5gb,iops=30
+_pv_vol_02                              RWO,ROX             size=5gb,iops=30
+_pv_vol_03          name=other-data     RWO,ROX             size=25gb,iops=30
+
+```
+
+#### Request persistent storage for a pod
+
+Susie is happy to see storage that can be mounted many times (ReadOnlyMany).  That satisfies her load balancer requirement.  Susie creates an pod image with her number crunching code and requests persistence for her volume.
+
+Susie's YAML looks something like this for the pod that creates her data:
+
+
+```yaml
+
+id: creator-of-data
+kind: Pod
+namespace: ns
+apiVersion: v1beta1
+desiredState:
+  manifest:
+    version: v1beta1
+    id: creator-of-data
+    containers:
+      - name: numbercruncher
+        image: ns/makesomdata:v1
+        command:
+          - /run.sh
+        volumeMounts:
+          - name: data
+            mountPath: /some_data
+    volumes:
+      - name: data
+        persistenceRequirements:
+        	mode: readonlymany
```

Comment:
Then how does the PersistentStoreController know to attach the volume as read-write to the creator-of-data pod so that it can fill the store with data?